gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/Context.java

```diff
@@ -219,8 +221,11 @@ private String determineMaterializeProperties() {
                         : Tokens.MATERIALIZE_PROPERTIES_ALL;
         }
 
+        final String materializeProperties = requestMessage.getField(Tokens.ARGS_MATERIALIZE_PROPERTIES);
         // all options except MATERIALIZE_PROPERTIES_TOKENS treated as MATERIALIZE_PROPERTIES_ALL
-        return Tokens.MATERIALIZE_PROPERTIES_ALL;
+        return (null != materializeProperties) && materializeProperties.equals(Tokens.MATERIALIZE_PROPERTIES_TOKENS)
```

Comment:
nit: can be  simplified to `Tokens.MATERIALIZE_PROPERTIES_TOKENS.equals(materializeProperties)`